tests/test_on_data/test_imaging_interfaces.py

```diff
@@ -228,3 +232,50 @@ def check_read_nwb(self, nwbfile_path: str):
             assert_array_equal(two_photon_series.dimension[:], self.two_photon_series_metadata["dimension"])
 
         super().check_read_nwb(nwbfile_path=nwbfile_path)
+
+
+class TestMiniscopeImagingInterface(ImagingExtractorInterfaceTestMixin, TestCase):
+    data_interface_cls = MiniscopeImagingInterface
+    interface_kwargs = dict(folder_path=str(OPHYS_DATA_PATH / "imaging_datasets" / "Miniscope" / "C6-J588_Disc5"))
+    save_directory = OUTPUT_PATH
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        cls.device_name = "Miniscope"
+        cls.imaging_plane_name = "ImagingPlane"
+        cls.photon_series_name = "OnePhotonSeries"
+
+        cls.device_metadata = dict(
+            compression="FFV1",
+            deviceType="Miniscope_V3",
+            frameRate="15FPS",
+            framesPerFile=1000,
+            gain="High",
+            led0=47,
+        )
+
+    def check_extracted_metadata(self, metadata: dict):
+        self.assertEqual(
+            metadata["NWBFile"]["session_start_time"],
+            datetime(2021, 10, 7, 15, 3, 28, 635),
+        )
+
+    def check_read_nwb(self, nwbfile_path: str):
+        with NWBHDF5IO(nwbfile_path, "r") as io:
+            nwbfile = io.read()
+
+            # Check device metadata
+            self.assertIn(self.device_name, nwbfile.devices)
+            device = nwbfile.devices[self.device_name]
+            self.assertIsInstance(device, Miniscope)
+            self.assertEqual(device.fields, self.device_metadata)
+            self.assertEqual(device, nwbfile.imaging_planes[self.imaging_plane_name].device)
+
+            # Check OnePhotonSeries
+            self.assertIn(self.photon_series_name, nwbfile.acquisition)
+            self.assertEqual(nwbfile.acquisition[self.photon_series_name].unit, "px")
+
+        imaging = self.interface.imaging_extractor
+        nwb_imaging = NwbImagingExtractor(file_path=nwbfile_path, optical_series_name="OnePhotonSeries")
```

Comment:
this also requires a refactor on the `roiextractors` side https://github.com/catalystneuro/roiextractors/pull/229